alibi/explainers/cfproto.py

```diff
@@ -734,6 +743,10 @@ def fit(self, train_data: np.ndarray, trustscore_kwargs: dict = None, d_type: st
 
             # combined distance measure
             if d_type == 'abdm-mvdm':
+                if w is None:
+                    msg = f"Must specify a value for `w` if using d_type='abdm-mvdm'"
+                    raise ValueError(msg)
```

Comment:
This validation was missing which the type-checker caught as trying to do arithmetic with `w=None` and integers later in the function.